NEWS.md

```diff
@@ -33,6 +33,11 @@ Breaking changes
 
 This section lists changes that do not have deprecation warnings.
 
+  * `String`s no longer have a `.data` field (as part of a significant performance
+    improvement). Use `Vector{UInt8}(str)` to access a string as a byte array.
```

Comment:
Probably mention `sizeof(str)` here too, since `length(str.data)` is pretty common.